lib/Controller/MessagesController.php

```diff
@@ -220,6 +220,55 @@ public function move($accountId, $folderId, $id, $destAccountId, $destFolderId):
 		return new JSONResponse();
 	}
 
+	/**
+	 * @NoAdminRequired
+ 	 * @NoCSRFRequired
+	 * @TrapError
+	 *
+	 * @param int $accountId
+	 * @param string $folderId
+	 * @param int $messageId
+	 *
+	 * @return JSONResponse
+	 * @throws ServiceException
+	 */
+	public function getRawMessage(int $accountId, string $folderId, int $messageId): Response {
+		try {
+			$account = $this->accountService->find($this->currentUserId, $accountId);
+		} catch (DoesNotExistException $e) {
+			return new TemplateResponse(
+				$this->appName,
+				'error',
+				['message' => 'Not allowed'],
+				'none'
+			);
+		}
+
+		$htmlResponse = new HtmlResponse(
+			nl2br(
+				$this->mailManager->getRawMessage(
```

Comment:
As Scrutinizer says: `The method ``getRawMessage()`` does not exist on ``OCA\Mail\Contracts\IMailManager``. Since it exists in all sub-types, consider adding an abstract or default implementation to ``OCA\Mail\Contracts\IMailManager``.`

Reply:
Could you please confirm that my change correctly addresses your concern?

Reply:
Yep, looks good now :)